app/models/street/StreetEdgePriorityTable.scala

```diff
@@ -149,6 +154,32 @@ object StreetEdgePriorityTable {
     }
   }
 
+  /**
+    * Recalculate the priority attribute for all streetEdges.
+    *
+    * @param rankParameterGeneratorList List of funcs that generate a number between 0 and 1 for each streetEdge.
+    * @param weightVector List of positive numbers b/w 0 and 1 that sum to 1; used to weight the generated parameters.
+    * @return
+    */
+  def updateAllStreetEdgePrioritiesTakeTwo(rankParameterGeneratorList: List[()=>List[StreetEdgePriorityParameter]], weightVector: List[Double]) = db.withTransaction { implicit session =>
```

Comment:
Could you explain what this function does?

Reply:
It does what `updateAllStreetEdgePriorities` does, which is compute a weighted sum of factors that influence priority. It takes a list of functions that generate a list of StreetEdgePriorityParameters (which just means a value between 0 and 1 representing priority for each street), and for each street edge, it computes a weighted sum of the priority parameters to get our final street edge priority.